pkg/pod/pod.go

```diff
@@ -43,19 +43,18 @@ func addPodHandler(i informersv1.PodInformer, eventCh chan aadpodid.EventType) {
 			AddFunc: func(obj interface{}) {
 				klog.V(6).Infof("pod created")
 				eventCh <- aadpodid.PodCreated
-
 			},
 			DeleteFunc: func(obj interface{}) {
 				klog.V(6).Infof("pod deleted")
 				eventCh <- aadpodid.PodDeleted
-
 			},
 			UpdateFunc: func(OldObj, newObj interface{}) {
-				// We are only interested in updates to pod if the node changes.
+				// We are only interested in updates to pod if the node or label changes.
 				// Having this check will ensure that mic sync loop does not do extra work
 				// for every pod update.
-				if (OldObj.(*v1.Pod)).Spec.NodeName != (newObj.(*v1.Pod)).Spec.NodeName {
-					klog.V(6).Infof("Pod Updated")
+				oldPod, newPod := OldObj.(*v1.Pod), newObj.(*v1.Pod)
```

Comment:
Did the old identity get removed and new identity assigned when the pod label was updated?

Reply:
Are you referring to the old AzureAssignedIdentity?

Reply:
Yeah, when the label is changed it should trigger deletion of old `AzureAssignedIdentity` and creation new one? Also the identity associated with the old `AzureIdentityBinding` label should be unassigned and new identity assigned. Wanted to confirm if you're seeing that behavior?

Reply:
I am seeing AzureAssignedIdentity being updated instead of being deleted and re-added:
```
I0714 19:17:39.808599       1 mic.go:1006] processing node k8s-agentpool1-95854893-vmss, add [0], del [0], update [1]
I0714 19:17:39.808660       1 crd.go:546] updating assigned id default/n0-default-demo
I0714 19:17:39.843077       1 crd.go:779] updating AzureAssignedIdentity default/n0-default-demo status to Assigned
I0714 19:17:39.851671       1 mic.go:511] work done: true. Found 1 pods, 1 ids, 2 bindings
I0714 19:17:39.851695       1 mic.go:512] total work cycles: 73, out of which work was done in: 3
I0714 19:17:39.851708       1 stats.go:99] ** stats collected **
I0714 19:17:39.851715       1 stats.go:82] Pod listing: 75.901µs
I0714 19:17:39.851727       1 stats.go:82] ID listing: 3.6µs
I0714 19:17:39.851734       1 stats.go:82] Binding listing: 8.5µs
I0714 19:17:39.851741       1 stats.go:82] Assigned ID listing: 10.6µs
I0714 19:17:39.851748       1 stats.go:82] System: 274.101µs
I0714 19:17:39.851755       1 stats.go:82] CacheSync: 4.3µs
I0714 19:17:39.851762       1 stats.go:82] Cloud provider get: 24.354029ms
I0714 19:17:39.851768       1 stats.go:82] Cloud provider put: 0s
I0714 19:17:39.851775       1 stats.go:82] Assigned ID addition: 9.980912ms
I0714 19:17:39.851783       1 stats.go:82] Assigned ID deletion: 0s
I0714 19:17:39.851790       1 stats.go:89] Number of cloud provider PUT: 0
I0714 19:17:39.851797       1 stats.go:89] Number of cloud provider GET: 1
I0714 19:17:39.851803       1 stats.go:89] Number of assigned ids created in this sync cycle: 0
I0714 19:17:39.851809       1 stats.go:89] Number of assigned ids updated in this sync cycle: 1
I0714 19:17:39.851815       1 stats.go:89] Number of assigned ids deleted in this sync cycle: 0
I0714 19:17:39.851822       1 stats.go:82] Find assigned ids to create: 4.2µs
I0714 19:17:39.851828       1 stats.go:82] Find assigned ids to delete: 3.6µs
I0714 19:17:39.851835       1 stats.go:82] Total time to assign or remove IDs: 43.060351ms
I0714 19:17:39.851842       1 stats.go:82] Event recording: 0s
I0714 19:17:39.851849       1 stats.go:82] Total: 43.555352ms
I0714 19:17:39.851856       1 stats.go:129] *********************
```

Reply:
Nevermind, I was using the same identity for both identity binding. Let me test it again

Reply:
Confirmed that the old AzureAssignedIdentity was removed and a new on is created:
`I0714 20:51:32.256057       1 mic.go:1006] processing node k8s-agentpool1-95854893-vmss, add [1], del [1], update [0]`